src/App.tsx

```diff
@@ -9,37 +9,84 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { User } from './types/User';
+import { Post } from './types/Post';
+
+import { getPosts } from './api/ApiMethods';
+
 export const App: React.FC = () => {
+  const [selectUser, setSelectUser] = useState<User | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasSelect, setHasSelect] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [openPost, setOpenPost] = useState<Post | null>(null);
+  const [openForm, setOpenForm] = useState(false);
```

Comment:
```suggestion
  const [isFormOpen, setIsFormOpen] = useState(false);
```
fix everywhere: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig